src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -79,65 +87,82 @@ public virtual void FileWorkItems(string sarifLogFileContents)
+                    dataInsertingVisitor.Visit(sarifLog);
                 }
-                case SplittingStrategy.PerResult:
+
+                SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
+
+                if (splittingStrategy == SplittingStrategy.None)
                 {
-                    partitionFunction = (result) => result.ShouldBeFiled() ? Guid.NewGuid().ToString() : null;
-                    break;
+                    FileWorkItemsHelper(sarifLog, this.FilingContext, this.FilingClient);
+                    return;
                 }
-                default:
+
+                IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+                PartitionFunction<string> partitionFunction = null;
+
+                Stopwatch splittingStopwatch = Stopwatch.StartNew();
+                
+                switch (splittingStrategy)
                 {
-                    throw new ArgumentOutOfRangeException($"SplittingStrategy: {splittingStrategy}");
+                    case SplittingStrategy.PerRun:
+                    {
+                        partitionFunction = (result) => result.ShouldBeFiled() ? "Include" : null;
+                        break;
+                    }
+                    case SplittingStrategy.PerResult:
+                    {
+                        partitionFunction = (result) => result.ShouldBeFiled() ? Guid.NewGuid().ToString() : null;
+                        break;
+                    }
+                    default:
+                    {
+                        throw new ArgumentOutOfRangeException($"SplittingStrategy: {splittingStrategy}");
+                    }
                 }
-            }
 
-            var partitioningVisitor = new PartitioningVisitor<string>(partitionFunction, deepClone: false);
-            partitioningVisitor.VisitSarifLog(sarifLog);
+                var partitioningVisitor = new PartitioningVisitor<string>(partitionFunction, deepClone: false);
+                partitioningVisitor.VisitSarifLog(sarifLog);
 
-            logsToProcess = new List<SarifLog>(partitioningVisitor.GetPartitionLogs().Values);
+                logsToProcess = new List<SarifLog>(partitioningVisitor.GetPartitionLogs().Values);
 
-            for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
-            {
-                SarifLog splitLog = logsToProcess[splitFileIndex];
-                FileWorkItemsHelper(splitLog, this.FilingContext, this.FilingClient);
+                var logsToProcessMetrics = new Dictionary<string, object>
+                {
+                    { "splittingStrategy", splittingStrategy },
+                    { "logsToProcessCount", logsToProcess.Count },
+                    { "splittingDurationInMilliseconds", splittingStopwatch.ElapsedMilliseconds },
+                };
+
+                this.Logger.LogMetrics(EventIds.LogsToProcessMetrics, logsToProcessMetrics);
```

Comment:
>this [](start = 16, length = 4)

@alisonmaclellan, this is how you would publish metrics to AI.

Reply:
@ So if I want to add more event ids, I'll add it to the EventIds.cs. 

---
In reply to: [395894423](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r395894423) [](ancestors = 395894423)

Reply:
Yes, I think that would be the plan. I don't think it makes sense to have a different EventIds.cs file per project since the logger is only used for work items. i.e. we're treating the WorkItems project as a 'core' project that can hold infrastructure object.

---
In reply to: [395897554](https://github.com/microsoft/sarif-sdk/pull/1820#discussion_r395897554) [](ancestors = 395897554,395894423)